src/handlers/check_commits/no_mentions.rs

```diff
@@ -1,37 +1,29 @@
 //! Purpose: When opening a PR, or pushing new changes, check for github mentions
 //! in commits and notify the user of our no-mentions in commits policy.
 
-use std::fmt::Write;
-
 use crate::{config::NoMentionsConfig, github::GithubCommit};
 
 pub(super) fn mentions_in_commits(
     _conf: &NoMentionsConfig,
     commits: &[GithubCommit],
 ) -> Option<String> {
-    let mut mentions_commits = Vec::new();
-
-    for commit in commits {
-        if !parser::get_mentions(&commit.commit.message).is_empty() {
-            mentions_commits.push(&*commit.sha);
-        }
-    }
+    let mentions_commits = commits
+        .into_iter()
+        .filter(|c| !parser::get_mentions(&c.commit.message).is_empty())
+        .map(|c| format!("    - {}\n", c.sha))
+        .collect::<String>();
 
     if mentions_commits.is_empty() {
         None
     } else {
-        Some(mentions_in_commits_warn(mentions_commits))
-    }
-}
-
-fn mentions_in_commits_warn(commits: Vec<&str>) -> String {
-    let mut warning = format!("There are username mentions (such as `@user`) in the commit messages of the following commits.\n  *Please remove the mentions to avoid spamming these users.*\n");
-
-    for commit in commits {
-        let _ = writeln!(warning, "    - {commit}");
+        Some(
+            String::from(
+                r"There are username mentions (such as `@user`) in the commit messages of the following commits.
+  *Please remove the mentions to avoid spamming these users.*
+",
+            ) + &mentions_commits,
+        )
     }
-
-    warning
 }
```

Comment:
@Kobzol, you had some opinions in https://github.com/rust-lang/triagebot/pull/1922#discussion_r2026800801, how about what's being proposed here?

Reply:
Performance is irrelevant here, this is not code is not hot. That being said, the iterator change looks pretty nice, and a separate function probably wasn't needed.

The second change is a bit weird though, it seems perplexing and unidiomatic to create a string and then  immediately add another one to it using `+`. We should just use `format!` there.